src/parsers/pm2ps_parser.jl

```diff
@@ -216,33 +216,42 @@ function make_ren_gen(gen_name, d, bus)
 end
 
 function make_thermal_gen(gen_name, d, bus)
-    if d["model"] ==1
+    model = GeneratorCostModel(d["model"])
+    if model == piecewise_linear::GeneratorCostModel
         cost_component = d["cost"]
         power_p = [i for (ix,i) in enumerate(cost_component) if isodd(ix)]
         cost_p =  [i for (ix,i) in enumerate(cost_component) if iseven(ix)]./power_p
         cost = [(p,c) for (p,c) in zip(cost_p,power_p)]
         fixedcost = cost[1][2]
-    elseif d["model"] ==2
-        if d["ncost"] == 2
+    elseif model == polynomial::GeneratorCostModel
+        if d["ncost"] == 0
+            cost = x-> 0
+        elseif d["ncost"] == 1
+            cost = x-> d["cost"][1]
+        elseif d["ncost"] == 2
             cost = x-> d["cost"][1]*x + d["cost"][2]
         elseif d["ncost"] == 3
             cost = x-> d["cost"][1]*x^2 + d["cost"][2]*x + d["cost"][3]
         elseif d["ncost"] == 4
             cost = x-> d["cost"][1]*x^3 + d["cost"][2]*x^2 + d["cost"][3]*x + d["cost"][4]
+        else
+            throw(DataFormatError("invalid value for ncost: $(d["ncost"])"))
         end
-        fixedcost = cost(0)
-    else
-        cost = d["cost"]
-        fixedcost = 0.0
+
+        # TODO: Reviewers:  Is this correct?
+        fixedcost = cost(d["pmin"])
     end
+
+    # GitHub #148: ramp_agc isn't always present. This value may not be correct.
+    ramp_agc = get(d, "ramp_agc", 0.0)
```

Comment:
@jd-lara Please comment on whether this is appropriate for now.

Reply:
This is fine for now since we don't have an AGC service yet. It might be worthwhile adding a TODO that potentially this value will be sent somewhere.